developer/frameworks/articles/data-engine/00-data-engine-intro.markdown

```diff
@@ -0,0 +1,46 @@
+# Data Engine
+
+Past versions of @product@ included a robust framework for creating form and
+list-based applications, either programmatically or from the Dynamic Data Lists
+Application. While Dynamic Data Lists continue to exist, a new framework was
+written to be based on the Forms Application's backend. ????? SAY WHY ?????. In
+@product-ver@, only the backend APIs are available for developers.
+
+The Data Engine framework has three fundamental entities. You'll be familiar
+with these entities if you used the Dynamic Data List framework in the past.
+
+1.  Data Definitions
+2.  Data Record Collections
+3.  Data Records
+
+##  Data Definitions
+
+Data Definitions (`DEDataDefinition`s) hold the form's field types (e.g., text,
+boolean, date, radio buttons, selector menus, etc.) and those fields' labels and
+settings. Data definitions effectively serve as data models for a Data
+Collection. For example, you could create a data definition with two text
+fields: one for a user's name, and one for their comments. You could then
+display a form that gathers user feedback via a dynamic data list that uses that
+data definition. 
+
+## Data Record Collections
+
+Data Record Collections (`DEDataRecordCollection`s) collect Data Records.
+They're used to associate a displayed form with the data records it holds and
+its backing data definition.
+
+## Data Records
+
+Data Records are the individual form entries for a given Data Record
+Collection.
+
+## Data Views
+
+There's another important piece to the Data Engine framework: displays.
+
+NOT WITH TEMPLATES. TALK TO JEYVISON. Once you have a Data Definition and a Data
```

Comment:
Hello :) 

So, for this first release, at least, the layout will be defined in rows and columns, the same way we do with forms today. I'm working in the Data Layout story right now. it will be more clear when you can see it.

To summarize, the layout will be built with pages, rows and columns, just like forms.

Reply:
Okay, so how about this?

> Once you have a Data Definition and a Data Record Collection, it's common to need control over the dis[play. Use the DDM Form annotations to construct and lay out the form display. The annotations are found in the `com.liferay.dynamic.data.mapping.annotations` package in the  `dynamic-data-mapping-api` module. For an example usage, look at the dynamic-data-mapping-form-field-type module's TextDDMFormFieldTypeSettings class, which defines the Text field's settings screen in the built-in Forms application.

Reply:
ugh, i hit "start review" accidentally, hope that doesn't mess you guys up.

Reply:
Ohhh, got it. My bad. I didn't explain it right XD  The developer will not use forms annotations. The row/column scheme will be the same but they will use the Data Layout entity for that( together with DataLayout{Page/Row/Column} You can see a [WIP](https://github.com/jeyvison/liferay-portal/pull/200) of the Data Layout here.

Reply:
Ohhh, I see, so DEDataLayout and the other layout concepts are separate model entities. Okay.

> Once you have a Data Definition and a Data Record Collection, it's common to need control over the display. Your Data Views can be comprised of several model entities representing these layout concepts:
> - The form layout
> - Form columns
> - Form row
> - Form pages

Reply:
Yeah that's it. I'd remove the form name too. Maybe Data Layout would suit better?